### DISCUSSION TITLE
Permissions for Platform Credential mode

### DISCUSSION THREAD

**Discussion by Golgo1** [mulesoft-catalyst/metrics-toolkit#93] [Q&A]
Is there any guidance on what permissions must be set within Access Management in order to use an account with Platform Credentials mode?

The table in README lists the Product type that can be queried, but no details on if any special permissions are required.
(Perhaps I have just missed noticing such details)

I have Org Administrator permission, but all my requests receive 401 unauth

**Comment by Golgo1** [comment]
To add a bit of detail, the 401 returns the specific  "error_description": "Invalid grant_type"

This is when attempting to get a token
https://anypoint.mulesoft.com/accounts/api/v2/oauth2/token

My headers consist of 
X-ANYPNT-ORG-ID
X-ANYPNT-CLIENT-SECRET
X-ANYPNT-CLIENT-ID
Providing my OrdID, and its ClientID/Secret

**Reply by sbegg** [reply]
Well, you have to add two more headers if you want to use platform credentials

- X-ANYPNT-USERNAME
- X-ANYPNT-PASSWORD

And don't forget to set the application property to indicate if you are using platform credentials or a connected app.

**Reply by santosh-sm** [reply]
Hi 

Although iam passing all the above headers, iam still facing same issue.

X-ANYPNT-ORG-ID
X-ANYPNT-CLIENT-SECRET
X-ANYPNT-CLIENT-ID
X-ANYPNT-USERNAME
X-ANYPNT-PASSWORD

and in application property 
```
auth:
  mode: "platform-credentials"
```

iam getting this below error.

```
{
    "message": "Internal Server error",
    "error": "HTTP POST on resource 'https://anypoint.mulesoft.com:443/accounts/api/v2/oauth2/token' failed: unauthorized (401)."
}
```

**Reply by richardmckinley** [reply]
Hi @santosh-sm 
I hope you have managed to resolve this. If not, please see the [requirements](https://github.com/mulesoft-catalyst/metrics-toolkit#requirements) section in the main documentation. 

There are 2 auth options available. 
Platform user- where you provide a platform user accounts username and password, and connected apps, where you provide the client ID and secret for a connected app. In both cases, you must ensure that the user or app has all required permissions in all business groups and environments. 

The headers for platform user are:

- X-ANYPNT-ORG-ID:{{orgId}}
- X-ANYPNT-USERNAME:{{username}}
- X-ANYPNT-PASSWORD:{{password}}
- X-PLAINTEXT-PASSWORD:false --(optional - defaults to false) 

The headers for connected app are: 

- X-ANYPNT-ORG-ID:{{orgId}}
- X-ANYPNT-CLIENT-ID:{{connectedAppId}}
- X-ANYPNT-CLIENT-SECRET:{{connectedAppSecret}}

Thanks, 
Richard
